test:TestCtlV3WatchTimeout

```
--- FAIL: TestCtlV3WatchTimeout (1.59s)
	ctl_v3_watch_test.go:74: watchTest #3-2: ctlV3Put error (read /dev/ptmx: input/output error (expected "OK", got ["Incorrect Usage.\r\n" "\r\n" "NAME:\r\n" "   etcdctl - A simple command line client for etcd.\r\n" "\r\n" "USAGE:\r\n" "   ../bin/etcdctl [global options] command [command options] [arguments...]\r\n" "   \r\n" "VERSION:\r\n" "   3.0.0-beta.0+git\r\n" "   \r\n" "COMMANDS:\r\n" "   backup\t\tbackup an etcd directory\r\n" "   cluster-health\tcheck the health of the etcd cluster\r\n" "   mk\t\t\tmake a new key with a given value\r\n" "   mkdir\t\tmake a new directory\r\n" "   rm\t\t\tremove a key or a directory\r\n" "   rmdir\t\tremoves the key if it is an empty directory or a key-value pair\r\n" "   get\t\t\tretrieve the value of a key\r\n" "   ls\t\t\tretrieve a directory\r\n" "   set\t\t\tset the value of a key\r\n" "   setdir\t\tcreate a new directory or update an existing directory TTL\r\n" "   update\t\tupdate an existing key with a given value\r\n" "   updatedir\t\tupdate an existing directory\r\n" "   watch\t\twatch a key for changes\r\n" "   exec-watch\t\twatch a key for changes and exec an executable\r\n" "   member\t\tmember add, remove and list subcommands\r\n" "   import\t\timport a snapshot to a cluster\r\n" "   user\t\t\tuser add, grant and revoke subcommands\r\n" "   role\t\t\trole add, grant and revoke subcommands\r\n" "   auth\t\t\toverall auth controls\r\n" "   help, h\t\tShows a list of commands or help for one command\r\n" "   \r\n" "GLOBAL OPTIONS:\r\n" "   --debug\t\t\toutput cURL commands which can be used to reproduce the request\r\n" "   --no-sync\t\t\tdon't synchronize cluster information before sending request\r\n" "   --output, -o \"simple\"\toutput response in the given format (`simple`, `extended` or `json`)\r\n" "   --discovery-srv, -D \t\tdomain name to query for SRV records describing cluster endpoints\r\n" "   --peers, -C \t\t\tDEPRECATED - \"--endpoints\" should be used instead\r\n" "   --endpoint \t\t\tDEPRECATED - \"--endpoints\" should be used instead\r\n" "   --endpoints \t\t\ta comma-delimited list of machine addresses in the cluster (default: \"http://127.0.0.1:2379,http://127.0.0.1:4001\")\r\n" "   --cert-file \t\t\tidentify HTTPS client using this SSL certificate file\r\n" "   --key-file \t\t\tidentify HTTPS client using this SSL key file\r\n" "   --ca-file \t\t\tverify certificates of HTTPS-enabled servers using this CA bundle\r\n" "   --username, -u \t\tprovide username[:password] and prompt if password is not supplied.\r\n" "   --timeout \"1s\"\t\tconnection timeout per request\r\n" "   --total-timeout \"5s\"\t\ttimeout for the command execution (except watch)\r\n" "   --help, -h\t\t\tshow help\r\n" "   --version, -v\t\tprint the version\r\n" "   \r\n"]))
```

--- a/e2e/ctl_v3_watch_test.go
+++ b/e2e/ctl_v3_watch_test.go
@@ -61,25 +61,28 @@ func watchTest(cx ctlCtx) {
 			[]kv{{"etcd", "revision_2"}, {"etcd", "revision_3"}},
 		},
 		{ // watch 3 keys by range
-			[]kv{{"key1", "val1"}, {"key2", "val2"}, {"key3", "val3"}},
+			[]kv{{"key1", "val1"}, {"key3", "val3"}, {"key2", "val2"}},
 			[]string{"key", "key3", "--rev", "1"},
 			[]kv{{"key1", "val1"}, {"key2", "val2"}},
 		},
 	}
 
 	for i, tt := range tests {
+		donec := make(chan struct{})
 		go func(i int, puts []kv) {
 			for j := range puts {
 				if err := ctlV3Put(cx, puts[j].key, puts[j].val, ""); err != nil {
 					cx.t.Fatalf("watchTest #%d-%d: ctlV3Put error (%v)", i, j, err)
 				}
 			}
+			close(donec)
 		}(i, tt.puts)
 		if err := ctlV3Watch(cx, tt.args, tt.wkv...); err != nil {
 			if cx.dialTimeout > 0 && !isGRPCTimedout(err) {
 				cx.t.Errorf("watchTest #%d: ctlV3Watch error (%v)", i, err)
 			}
 		}
+		<-donec
 	}
 }
 

